src/Resource.ts

```diff
@@ -19,6 +20,7 @@ export interface ShorthandResourceActionOptions {
   action: string
   property?: string
   path: Function | string
+  autoCancel?: Boolean
```

Comment:
I think we need an `onCancel` option as well; cleanup can be quite different depending on the situation.